omnigibson/scenes/interactive_traversable_scene.py

```diff
@@ -179,7 +179,7 @@ def _should_load_object(self, obj_info):
         valid_room = self.load_room_instances is None or len(set(self.load_room_instances) & set(in_rooms)) > 0
 
         # Check whether this is an agent and we allow agents
-        agent_ok = self.include_robots or category != robot_macros.ROBOT_CATEGORY
+        agent_ok = self.include_robots or (category != robot_macros.ROBOT_CATEGORY and obj_info["class_name"] not in REGISTERED_ROBOTS)
```

Comment:
hmm sorry what is this line doing?

Reply:
in the new RC we actually shouldn't run into this problem, but previously we weren't storing object categories if they weren't explicitly passed into the object constructor. So for robots, which have (internally) a hardcoded category of "agent", category wasn't getting stored (and the fallback in the code snippet above falls back to category `object`), so we could never filter out the robot because (`agent_ok` was always `True`).

This just adds an extra sanity check to make sure an actual robot class is not being loaded if we don't want to include_robots

Reply:
Got it.